light_sdk/src/lib.rs

```diff
@@ -0,0 +1,284 @@
+use std::collections::BTreeMap;
+use namada_core::proto::Section;
+
+/// Initialize account transaction WASM
+pub const TX_INIT_ACCOUNT_WASM: &str = "tx_init_account.wasm";
+/// Initialize validator transaction WASM path
+pub const TX_INIT_VALIDATOR_WASM: &str = "tx_init_validator.wasm";
+/// Unjail validator transaction WASM path
+pub const TX_UNJAIL_VALIDATOR_WASM: &str = "tx_unjail_validator.wasm";
+/// Deactivate validator transaction WASM path
+pub const TX_DEACTIVATE_VALIDATOR_WASM: &str = "tx_deactivate_validator.wasm";
+/// Reactivate validator transaction WASM path
+pub const TX_REACTIVATE_VALIDATOR_WASM: &str = "tx_reactivate_validator.wasm";
+/// Initialize proposal transaction WASM path
+pub const TX_INIT_PROPOSAL: &str = "tx_init_proposal.wasm";
+/// Vote transaction WASM path
+pub const TX_VOTE_PROPOSAL: &str = "tx_vote_proposal.wasm";
+/// Reveal public key transaction WASM path
+pub const TX_REVEAL_PK: &str = "tx_reveal_pk.wasm";
+/// Update validity predicate WASM path
+pub const TX_UPDATE_ACCOUNT_WASM: &str = "tx_update_account.wasm";
+/// Transfer transaction WASM path
+pub const TX_TRANSFER_WASM: &str = "tx_transfer.wasm";
+/// IBC transaction WASM path
+pub const TX_IBC_WASM: &str = "tx_ibc.wasm";
+/// User validity predicate WASM path
+pub const VP_USER_WASM: &str = "vp_user.wasm";
+/// Validator validity predicate WASM path
+pub const VP_VALIDATOR_WASM: &str = "vp_validator.wasm";
+/// Bond WASM path
+pub const TX_BOND_WASM: &str = "tx_bond.wasm";
+/// Unbond WASM path
+pub const TX_UNBOND_WASM: &str = "tx_unbond.wasm";
+/// Withdraw WASM path
+pub const TX_WITHDRAW_WASM: &str = "tx_withdraw.wasm";
+/// Claim-rewards WASM path
+pub const TX_CLAIM_REWARDS_WASM: &str = "tx_claim_rewards.wasm";
+/// Bridge pool WASM path
+pub const TX_BRIDGE_POOL_WASM: &str = "tx_bridge_pool.wasm";
+/// Change commission WASM path
+pub const TX_CHANGE_COMMISSION_WASM: &str =
+    "tx_change_validator_commission.wasm";
+/// Change consensus key WASM path
+pub const TX_CHANGE_CONSENSUS_KEY_WASM: &str = "tx_change_consensus_key.wasm";
+/// Change validator metadata WASM path
+pub const TX_CHANGE_METADATA_WASM: &str = "tx_change_validator_metadata.wasm";
+/// Resign steward WASM path
+pub const TX_RESIGN_STEWARD: &str = "tx_resign_steward.wasm";
+/// Update steward commission WASM path
+pub const TX_UPDATE_STEWARD_COMMISSION: &str =
+    "tx_update_steward_commission.wasm";
+/// Redelegate transaction WASM path
+pub const TX_REDELEGATE_WASM: &str = "tx_redelegate.wasm";
+/// Target chain ID
+pub const CHAIN_ID: &str = "localnet.3e837878d84b54a40f-0";
+/// Reveal public key transaction code hash
+pub const TX_REVEAL_PK_HASH: [u8; 32] = [0x8d, 0x4e, 0x20, 0x08, 0x30, 0xa9, 0xf6, 0x03, 0xd6, 0x47, 0x79, 0x9f, 0xaf, 0xdb, 0xf1, 0x94, 0x6e, 0x07, 0xa0, 0x38, 0xa6, 0x7c, 0xfa, 0x09, 0xd4, 0xee, 0xb3, 0x1c, 0x80, 0xc7, 0x74, 0xce];
+
+/// Takes any kind of inner tx and gives me back my sign bytes
+pub fn get_inner_sign_bytes(mut tx: Tx) -> Hash {
```

Comment:
These should be immutable functions on a specific RevealPk struct. Ie. this signature should be repeated for every type of transaction.

Also this function should be immutable.